tests/boardfarm_plugins/boardfarm_prplmesh/devices/prplmesh_prplwrt.py

```diff
@@ -27,29 +35,57 @@ def __init__(self, *args, **kwargs):
         self.args = args
         self.kwargs = kwargs
         config = kwargs.get("config", kwargs)
+
+        self.unique_id = os.getenv("SUDO_USER", os.getenv("USER", ""))
         self.docker_network = config.get("docker_network",
                                          "prplMesh-net-{}".format(self.unique_id))
         self.role = config.get("role", "agent")
-        if self.role == "controller":
-            self.controller_entity = ALEntityPrplWrt(self.name, is_controller=True)
-        else:
-            self.agent_entity = ALEntityPrplWrt(self.name, is_controller=False)
+        self.connection_type = config.get("connection_type", None)
+        self.conn_cmd = config.get("conn_cmd", None)
+
+        self.connection = connection_decider.connection(device=self,
+                                                        conn_type=kwargs['connection_type'],
+                                                        **kwargs)
+        self.connection.connect()
 
+        self.name = "-".join((config.get("name", "rax40"), self.unique_id))
+        self.consoles = [self]
+        self.logfile_read = sys.stdout
+
+        self.wan_network = self.get_docker_subnet(self.docker_network)
+        self.set_iface_to_bridge(self.wan_iface)
+        self.set_iface_ip("br-lan", self.wan_network)
```

Comment:
> I guess it's because you want to use an IP in the same subnet as the controller running in a docker container?

Exactly.

> This looks unnecessarily restrictive to me. It means we won't be able to run tests with a rax40 unless the controller is running in docker.

Initially, it was planned to have one spawned controller per DUT. Currently, boardfarm is unable to handle non-dockerized controller. We can create a separate boardfarm "device" for it.